src/mode.js

```diff
@@ -13,16 +13,25 @@ const chromaticNotes = ['c', 'c#', 'd', 'd#', 'e', 'f', 'f#', 'g', 'g#', 'a', 'a
  * @return {[type]}                        Returns the mode as an array, e.g. ['c3', 'd3', 'e3', 'f3', 'g3', 'a3', 'b3', 'c4']
  */
 const mode = (root, mode, octave, addRootFromNextOctave) => {
-	root = root || 'c';
-	mode = mode || 'ionian';
-	octave = octave ? Number(octave) : 3;
-	addRootFromNextOctave = addRootFromNextOctave === false ? false : true;
+	if (root.match(/^[A-Za-z]\|[A-Za-z]+/)) {
+		var args = root.split('|');
+		root = args[0];
+		mode = args[1];
+		octave = args[2];
+		addRootFromNextOctave = args[3] === undefined ? true : args[3] === 'true' ? true : false;
+	}
+
 	// Make sure the root is valid [abcdefg] optionally followed by #
 	assert(root.match(/[abcdefg]#?/i), 'Invalid root note: ' + root);
 
 	// Make sure if the provided mode is valid
 	assert(modes.hasOwnProperty(mode), 'Invalid mode: ' + mode);
 
+	root = root || 'c';
+	mode = mode || 'ionian';
+	octave = octave ? Number(octave) : 4;
+	addRootFromNextOctave = addRootFromNextOctave == false ? false : true;
```

Comment:
Once you do the above mentioned 2 changes pertaining to `addRootFromNextOctave` then you might not need this line anymore.